### PR TITLE
Add virtIO block driver and block example

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
This PR adds a virtIO block driver, which will allow us to easily use and test the block sub-system of sDDF as simulators such as QEMU expose a virtIO device. With the virtIO net driver we added recently, this will let us do a decent amount of experimentation in simulators.

The status is that the driver is functional and can read/write to a virtual disk, but more testing needs to be done. The client program is fairly simple right now, I'd like to combine it with the example in https://github.com/au-ts/sddf/pull/157 to have one example that is able to test block devices. This example does not connect to the virtualiser yet, which needs to be done as well.

To merge:
* [x] Add Makefile to example
~~* [ ] Add more extensive testing to example~~ for another time.
* [x] Add README to example
* [x] Connect virt to example

### CODE REVIEW

**Review by erichchan999 — DISMISSED** [review]

**Comment by Ivan-Velickovic** [comment]
One thing missing is that my driver doesn't respect the plugged functionality of the block interface.

**Comment by erichchan999** [comment]
Yep, though we probably won't have a proper driver anytime soon that implements the plug the interface (the virtualiser has ignored that for now too)

File: examples/blk/Makefile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#
```
[the remark this answers is not in the document]
Comment (Ivan-Velickovic) [reply examples/blk/Makefile:1]: I'm going to merge them later.

[the remark this answers is not in the document]
Comment (Ivan-Velickovic) [reply examples/blk/Makefile:1]: Right now my priority is having the driver in the repo.

[the remark this answers is not in the document]
Comment (Ivan-Velickovic) [reply examples/blk/Makefile:1]: Although for now I should rename `examples/blk/mcc` to `examples/mmc`.

[the remark this answers is not in the document]
Comment (Ivan-Velickovic) [reply examples/blk/Makefile:1]: Thanks for bringing that up, I forgot to make that change after rebasing.

File: examples/blk/board/qemu_virt_aarch64/blk.system [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+<!--
+ Copyright 2024, UNSW
+
+ SPDX-License-Identifier: BSD-2-Clause
+-->
+<system>
+    <!-- Device registers -->
+    <memory_region name="blk_regs" size="0x10_000" phys_addr="0xa003000" />
+    <!-- Needed specifically for the virtIO block driver -->
+    <!-- TODO: make this one region -->
```
Comment (Ivan-Velickovic) [inline examples/blk/board/qemu_virt_aarch64/blk.system:11]: Last thing to do before merging.

**Review by erichchan999 — APPROVED** [review]

**Pull request merged**